Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePage.h

```diff
@@ -27,6 +27,7 @@
 #define WKBundlePage_h
 
 #include <JavaScriptCore/JavaScript.h>
+#include <WebCore/CaptionUserPreferences.h>
```

Comment:
On Apple platforms, we treat WebCore as an implementation detail and do not support _clients_ of WebKit including WebCore headers. This means that any headers which go into the SDK (these are marked "public" and "private" headers in Xcode) need to avoid including WebCore, and I think it's why your iOS/tvOS/watchOS builds are failing.

Instead of directly including, you should try adding a line for `CaptionUserPreferences.h` to Source/WebKit/MigratedHeaders-input.xcfilelist and Source/WebKit/MigratedHeaders-output.xcfilelist. There's a build script that copies these headers into WebKit proper, and fixes them up so that they are importable without a bunch of WebCore types.

Reply:
and once you've added the line to the MigratedHeaders xcfilelists, you'd change this include statement to be
```suggestion
#include <WebKit/CaptionUserPreferences.h>
```

Reply:
Thank you for your advice. As per your suggestion, I have tried to add a line for `CaptionUserPreferences.h` to `MigrateHeaders-input/output.xcfilelist`. Then I changed the include statement. However, after making these changes, I encountered a series of errors related to the `namespace` and `using` keywords not being recognized. It seems like the C++ headers are being compiled as C code. I'm not sure how to proceed from here and would appreciate any further guidance.

Reply:
Ah, yes, this is C API.

Maybe you can find a simpler way to do this, but I think you could create an opaque type for CaptureUserPreferencesTestingModeToken in `WKBase.h`, add a mapping in `WKAPICast.h`, and a C API to create an opaque token. Then the test runner wouldn't need to know about the (WebCore) C++ types at all.

Reply:
I've implemented a type mapping for CaptureUserPreferencesTestingModeToken so that C API can handle it without knowing the implementation details in WebCore. Could you review it again? Thank you.